salt/tests/unit/modules/test_metalk8s_kubernetes_cronjob.py

```diff
@@ -0,0 +1,265 @@
+
+            self.assertEqual(
+                metalk8s_kubernetes_cronjob._set_cronjob_suspend(
+                    name, namespace, suspend
+                ),
+                result,
+            )
+
+    @utils.parameterized_from_cases(YAML_TESTS_CASES["suspend_cronjob"])
+    def test_suspend_cronjob(
+        self,
+        result,
+        name="my_cronjob",
+        namespace="default",
+    ):
+        mock__set_cronjob_suspend = MagicMock(return_value=True)
+
+        with patch.object(
+            metalk8s_kubernetes_cronjob,
+            "_set_cronjob_suspend",
+            mock__set_cronjob_suspend,
+        ):
+
+            self.assertEqual(
+                metalk8s_kubernetes_cronjob.suspend_cronjob(name, namespace), result
+            )
+
+    @utils.parameterized_from_cases(YAML_TESTS_CASES["activate_cronjob"])
+    def test_activate_cronjob(
+        self,
+        result,
+        name="my_cronjob",
+        namespace="default",
+    ):
+        mock__set_cronjob_suspend = MagicMock(return_value=True)
+
+        with patch.object(
+            metalk8s_kubernetes_cronjob,
+            "_set_cronjob_suspend",
+            mock__set_cronjob_suspend,
+        ):
+            self.assertEqual(
+                metalk8s_kubernetes_cronjob.activate_cronjob(name, namespace),
+                result,
+            )
+
+    @utils.parameterized_from_cases(YAML_TESTS_CASES["get_jobs"])
+    def test_get_jobs(
+        self,
+        result,
+        cronjob_name,
+        namespace,
+        list_objects=None,
+        get_cronjob=None,
+    ):
+        mock_list_objects = MagicMock(return_value=list_objects)
+        mock_get_cronjob = MagicMock(return_value=get_cronjob)
+
+        with patch.dict(
+            metalk8s_kubernetes_cronjob.__salt__,
+            {
+                "metalk8s_kubernetes.list_objects": mock_list_objects,
+                "metalk8s_kubernetes.get_object": mock_get_cronjob,
+            },
+        ), patch.object(
+            metalk8s_kubernetes_cronjob,
+            "get_cronjob",
+            mock_get_cronjob,
+        ):
+            self.assertEqual(
+                metalk8s_kubernetes_cronjob.get_jobs(cronjob_name, namespace),
+                result,
+            )
+
+    @utils.parameterized_from_cases(YAML_TESTS_CASES["stop_jobs"])
+    def test_stop_jobs(
+        self,
+        result,
+        cronjob_name,
+        namespace,
+        wait=False,
+        timeout_seconds=60,
+        get_jobs=None,
+        delete_jobs_delay=0,
+        raises=False,
+    ):
+        def mock_get_jobs():
+            launch_ts = time.time()
+
+            def get_jobs_call(cronjob_name, namespace, **kwargs):
+                if wait and time.time() - launch_ts >= delete_jobs_delay:
+                    return []
+                return get_jobs
+
+            return get_jobs_call
+
+        mock_delete_object = MagicMock(return_value=None)
+        mock_get_jobs = MagicMock(side_effect=mock_get_jobs())
```

Comment:
Could be simplified a bit (also you redefine `mock_get_jobs` here

```suggestion
        launch_ts = time.time()

        def get_jobs_call(cronjob_name, namespace, **kwargs):
            if wait and time.time() - launch_ts >= delete_jobs_delay:
                return []
            return get_jobs

        mock_delete_object = MagicMock(return_value=None)
        mock_get_jobs = MagicMock(side_effect=get_jobs_call)
```